tracer/src/Datadog.Trace/Agent/Transports/UdsClientRequestFactory.cs

```diff
@@ -0,0 +1,77 @@
+﻿// <copyright file="UdsClientRequestFactory.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET5_0_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.Agent.Transports
+{
+    internal class UdsClientRequestFactory : IApiRequestFactory
```

Comment:
Shouldn't we implement this like we did `HttpStreamRequestFactory` (passing a `IStreamFactory`) so it can be used for both UDS and named pipes?

Reply:
Probably? 😄

Reply:
I might do that in a separate PR though, as we don't have support for named pipes in our mock agent yet, so I'm a bit hesitant to make these changes 😬 The work is [here](https://github.com/DataDog/dd-trace-dotnet/commit/06353d1da1d51bbca7f3560eb327dbe326b4bdcf), but it's probably worth merging this PR first, given the other really requires extensive work on the `MockTracerAgent`?

Reply:
Sounds good.

Reply:
Actually, I pulled _most_ of that work into this branch, as doesn't seem worth keeping this intermediate step for now. The only thing I left out was actually using named pipes over HttpClient. I'll add support to the `MockTracerAgent` before we do that

Reply:
Thanks!